src/Agent.Listener/Configuration/AutoLogonRegistryManager.cs

```diff
@@ -0,0 +1,339 @@
+            }
+            else
+            {
+                screenSaverValue = _registryManager.GetValue(RegistryHive.CurrentUser, RegistryConstants.SubKeys.ScreenSaverDomainPolicy, RegistryConstants.ValueNames.ScreenSaver);
+            }
+
+            if (int.TryParse(screenSaverValue, out int isScreenSaverDomainPolicySet)
+                    && isScreenSaverDomainPolicySet == 1)
+            {
+                warningReasons.Add(StringUtil.Loc("UITestingWarning_ScreenSaver"));
+            }
+
+            return warningReasons;
+        }
+
+        private List<string> GetWarningsForMachineSpecificSettings()
+        {
+            var warningReasons = new List<string>();
+
+            //shutdown reason
+            var shutdownReasonValue = _registryManager.GetValue(RegistryHive.LocalMachine, RegistryConstants.SubKeys.ShutdownReasonDomainPolicy, RegistryConstants.ValueNames.ShutdownReason);
+            if (int.TryParse(shutdownReasonValue, out int shutdownReasonOn) 
+                    && shutdownReasonOn == 1)
+            {
+                warningReasons.Add(StringUtil.Loc("UITestingWarning_ShutdownReason"));
+            }
+
+            //legal caption/text
+            var legalNoticeCaption = _registryManager.GetValue(RegistryHive.LocalMachine, RegistryConstants.SubKeys.LegalNotice, RegistryConstants.ValueNames.LegalNoticeCaption);
+            var legalNoticeText =  _registryManager.GetValue(RegistryHive.LocalMachine, RegistryConstants.SubKeys.LegalNotice, RegistryConstants.ValueNames.LegalNoticeText);
+            if (!string.IsNullOrEmpty(legalNoticeCaption) || !string.IsNullOrEmpty(legalNoticeText))
+            {
+                warningReasons.Add(StringUtil.Loc("UITestingWarning_LegalNotice"));
+            }
+
+            //auto-logon
+            var autoLogonCountValue = _registryManager.GetValue(RegistryHive.LocalMachine, RegistryConstants.SubKeys.AutoLogon, RegistryConstants.ValueNames.AutoLogon);
+            if (!string.IsNullOrEmpty(autoLogonCountValue))
+            {
+                warningReasons.Add(StringUtil.Loc("UITestingWarning_AutoLogonCount"));
+            }           
+            
+            return warningReasons;
+        }
+
+        private void UpdateMachineSpecificRegistrySettings(string domainName, string userName)
+        {
+            var hive = RegistryHive.LocalMachine;
+
+            // SetValue(RegistryHive targetHive, string subKeyName, string name, string value)
+            SetValue(hive, RegistryConstants.SubKeys.AutoLogon, RegistryConstants.ValueNames.AutoLogonUserName, userName);
+            SetValue(hive, RegistryConstants.SubKeys.AutoLogon, RegistryConstants.ValueNames.AutoLogonDomainName, domainName);
+
+            //this call is to take the backup of the password key if already exists as we delete the key in the next step
+            SetValue(hive, RegistryConstants.SubKeys.AutoLogon, RegistryConstants.ValueNames.AutoLogonPassword, "");
```

Comment:
it was only to trigger the backup.

---
In reply to: [118825715](https://github.com/Microsoft/vsts-agent/pull/985#discussion_r118825715) [](ancestors = 118825715)